compiler/test/input/recursive-equal.gr

```diff
@@ -25,22 +28,23 @@ u := Some(w)
 
 assert x == w
 
+# These succeed with mutable record props
```

Comment:
Excellent to know. 😂 

Maybe it would be good to duplicate this test with both a box version and a mutable record field version so both code paths are exercised.

Reply:
👍 see above